linera-storage-service/src/server.rs

```diff
@@ -126,21 +137,68 @@ impl SharedStoreServer {
 
     pub async fn clear_journal(&self, base_key: &[u8]) -> Result<(), Status> {
         match self {
-            SharedStoreServer::Memory(store) => store
+            ServiceStoreServer::Memory(store) => store
                 .clear_journal(base_key)
                 .await
                 .map_err(|_e| Status::not_found("clear_journal")),
             #[cfg(feature = "rocksdb")]
-            SharedStoreServer::RocksDb(store) => store
+            ServiceStoreServer::RocksDb(store) => store
                 .clear_journal(base_key)
                 .await
                 .map_err(|_e| Status::not_found("clear_journal")),
         }
     }
+
+    pub async fn list_all(&self) -> Result<Vec<Vec<u8>>, Status> {
+        self.find_keys_by_prefix(&[KeyTag::Namespace as u8]).await
+    }
+
+    pub async fn delete_all(&self) -> Result<(), Status> {
+        let mut batch = Batch::new();
+        batch.delete_key_prefix(vec![KeyTag::Key as u8]);
+        batch.delete_key_prefix(vec![KeyTag::Namespace as u8]);
+        let base_key = vec![];
+        self.write_batch(batch, &base_key).await?;
+        self.clear_journal(&base_key).await
+    }
+
+    pub async fn exist_namespace(&self, namespace: &[u8]) -> Result<bool, Status> {
```

Comment:
Maybe this should be called `exists_namespace`. (Like e.g. `Path::exists` in the standard library.)

Reply:
Ok, done.